lending_pool/src/factory.rs

```diff
@@ -26,20 +29,59 @@ pub trait FactoryModule {
         arg_buffer.push_arg(r_slope2);
         arg_buffer.push_arg(u_optimal);
         arg_buffer.push_arg(reserve_factor);
+        arg_buffer.push_arg(liquidation_threshold);
 
-        let (address, _) = self.raw_vm_api().deploy_contract(
+        let (new_address, _) = self.raw_vm_api().deploy_from_source_contract(
             self.blockchain().get_gas_left(),
-            &amount,
-            bytecode,
-            code_metadata,
+            &BigUint::zero(),
+            &self.liq_pool_template_address().get(),
+            CodeMetadata::UPGRADEABLE,
             &arg_buffer,
         );
```

Comment:
Instead of manually building the calldata, you can use a contract proxy:
```
pub mod liq_pool_proxy {
    elrond_wasm::imports!();

    #[elrond_wasm_derive::proxy]
    pub trait LiqPoolProxy {
        #[init]
    fn init(
        &self,
        asset: TokenIdentifier,
        r_base: BigUint,
        r_slope1: BigUint,
        r_slope2: BigUint,
        u_optimal: BigUint,
        reserve_factor: BigUint,
        liquidation_threshold: BigUint,
    )
    }
}
```

Proxy object declaration:
```
#[proxy]
    fn liq_pool_proxy_obj(&self, sc_address: ManagedAddress) -> liq_pool_proxy::Proxy<Self::Api>;
```
And usage:
```
let address = self
            .liq_pool_proxy_obj(ManagedAddress::zero())
            .init(*arguments here*)
            .with_gas_limit(gas_per_deploy) <--- remove if you want to use all the remaining gas
            .deploy_from_source(&source_address, CodeMetadata::UPGRADEABLE);
```

Reply:
Both funcs are ok IMO. Both are exported by rust framework. The one you prefer comes with proxy declaration and much more overhead. I definitely prefer the raw_vm_api one. Same for below.